src/features/Header/Header.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { useNavigate } from "react-router";
+import { useAuth } from "../../context/authContext";
+import { useMutation } from "react-query";
+import { authSignOut } from "../../firebase/auth";
+import { Button } from "@mui/material";
+
+const Header = () => {
+  const navigate = useNavigate();
+  const { isLoggedIn } = useAuth();
+
+  const { mutate: signOut } = useMutation(authSignOut, {
+    onSuccess: () => {
+      navigate("/login");
+    },
+  });
+
+  const handleSignOut = () => {
+    signOut();
+  };
+
+  return (
+    <>
+      {isLoggedIn ? (
+        <>
+          <Button
+            onClick={handleSignOut}
+            variant="outlined"
+            disabled={signOut.isLoading}
+          >
+            {signOut.isLoading ? "Logging out..." : "Logout"}
```

Comment:
You can destructure also the isLoading parameter from the mutation.

```suggestion
  const { mutate: signOut, isLoading: isSignOutLoading } = useMutation(authSignOut, {
    onSuccess: () => {
      navigate("/login");
    },
  });

  const handleSignOut = () => {
    signOut();
  };

  return (
    <>
      {isLoggedIn ? (
        <>
          <Button
            onClick={handleSignOut}
            variant="outlined"
            disabled={signOut.isLoading}
          >
            {isSignOutLoading ? "Logging out..." : "Logout"}
```